### PR TITLE
Fix test fixtures

### LINKED ISSUE #488: Test databse inconsistent
Problem:
1) The test database has `user_id = 0` for the comments model whereas there is no User with `id = 0`. This inconsistency can create problems while writing tests
2) The test database has one `User.first` 's `tag.spectrum_id = 1` , but none of the spectra have `id = 1` 

Solution:
change the required fields in the test database seed.

@publiclab/community-reps , I would like to know your views on this

### PR DESCRIPTION
**Pull request by gr455**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!
Fixes #488

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by gr455** [comment]
Restarting Travis

File: test/fixtures/tags.yml [hunk 1]
```diff
@@ -6,12 +6,12 @@
 #
 one: {
  name: 'cfl',
- spectrum_id: 1,
+ spectrum_id: 298486374,
```
Comment (alaxalves) [inline test/fixtures/tags.yml]: Are you sure this id is static?

Comment (gr455) [reply test/fixtures/tags.yml]: You're right it's not. Thanks for pointing out, any suggestions to make the fixtures consistent without constant spectrum id?

Comment (alaxalves) [reply test/fixtures/tags.yml:9]: Get the last spectrum instance created and access it with `Spectrum.last`, or something like that. You have to make sure the the Spectrum fixtures is run first than the tags fixtures.

Comment (gr455) [reply test/fixtures/tags.yml:9]: I think it's easier to just set static the spectrum ids if it doesn't break other tests

Comment (gr455) [reply test/fixtures/tags.yml:9]: On a second thought, are they static, these numbers seem to repeat for most models (sets,comments,spectra...)

**Comment by gr455** [comment]
> looks good :+1: 

@cesswairimu can you confirm if the ids of the models are static or not in the fixtures

**Comment by cesswairimu** [comment]
@gr455 sorry, I didn't quite understand your request

**Comment by gr455** [comment]
> @gr455 sorry, I didn't quite understand your request

The models' ids in the test fixtures are some 9 digit numbers. These seem to be common for Spectra, sets, comments etc. But they're such wierd numbers. I was wondering if they have been assigned that way or are they  might differ depending on something (will they always be the same in the fixtures?)

**Comment by cesswairimu** [comment]
oh yeah, whatever you have in fixtures will be loaded as test data regardless of the environment..

**Comment by gr455** [comment]
> oh yeah, whatever you have in fixtures will be loaded as test data regardless of the environment..

But the ids aren't explicitly mentioned in the fixtures

**Comment by cesswairimu** [comment]
ooh yeah I see..
What you could do instead of explicitly defining the `spectrum_id`, have it as the name of the spectrum/user in the spectrum and user fixtures respectively ...e.g
```
one: {
 name: 'cfl',
 spectrum: one
 spectrum:  two,
}
........
  email: MyString
  body: MyText
  wavelength: 1.5
  user: quentin
  x: 1
  y: 1
```
Makes sense?

**Comment by gr455** [comment]
> ooh yeah I see..
> What you could do instead of explicitly defining the `spectrum_id`, have it as the name of the spectrum/user in the spectrum and user fixtures respectively ...e.g
> ```
> one: {
>  name: 'cfl',
>  spectrum: one
>  spectrum:  two,
> }
> ........
>   email: MyString
>   body: MyText
>   wavelength: 1.5
>   user: quentin
>   x: 1
>   y: 1
> ```
> Makes sense?

But then I'll have to make sure that spectrum fixtures load before tags

**Comment by cesswairimu** [comment]
I think all the fixtures required are loaded before the tests are run...or were you having trouble with this?

**Comment by gr455** [comment]
> I believe all the fixtures required are loaded before the tests are run...or were you having trouble with this? 

I tried something like that and got a 'method called on nil class' error, so i reckon that was the problem

**Comment by cesswairimu** [comment]
I thought that should work :thinking: Then I guess we can leave it as is and we can circle back if any errors occur

**Pull request merged**